server/client_test.go

```diff
@@ -384,13 +398,19 @@ func TestClientNoBodyPubSubWithReply(t *testing.T) {
 	}
 }
 
-func (c *client) parseFlushAndClose(op []byte) {
+// This needs to clear any flushOutbound flags since writeLoop not running.
+func (c *client) parseAndFlush(op []byte) {
 	c.parse(op)
 	for cp := range c.pcd {
 		cp.mu.Lock()
+		cp.flags.clear(flushOutbound)
```

Comment:
I would like to understand why we need this? The only place this flag is set is in flushOutbound() itself and is cleared when this function returns. The purpose of the flag was to prevent re-entrant calls since flushOutbound() releases client's lock when doing actual write to socket.

Reply:
These tests do not run a flush Go routine, and by default data (msgs) are expected to be sent via that Go routine, so we need to flush the pcd list in the test, which we do. What I was finding was that the flag was set every once in while so the manual flush was failing, so I clear the flag to force it in place.